pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -228,4 +240,38 @@ private static Key extraRowKey(Object[] row, List<RexExpression> groupSet) {
     }
     return new Key(keyElements);
   }
+
+  private static class Holder {
```

Comment:
minor refactor so we don't need to maintain three different arrays (including `Map[]`, which is generally considered bad practice)

Reply:
ah 👍 I think this clarifies my question above. yeah I think that's a good refactor, I'll do that